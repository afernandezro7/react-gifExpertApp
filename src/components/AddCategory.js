import React, { useState } from 'react';
import PropTypes from 'prop-types';



export const AddCategory = ({setCategories}) => {

    const [inputValue, setinputValue] = useState('Introduce un Elemento')

    const handleInputChange= (e)=>{
        setinputValue(e.target.value);
    }

    const handleKeyEnter= (e)=>{
        e.preventDefault();

        if( inputValue.trim().length > 2){
            
            setCategories(cats=>[inputValue,...cats])
            setinputValue('Introduce un Elemento')
            document.getElementById('Myform').reset()
        }
    }

    return (
        
        <form id="Myform" onSubmit= { handleKeyEnter }>
            <input
                type="text"               
                placeholder={inputValue}
                
                onChange={ handleInputChange }
            />
        </form>
        
    )
}

AddCategory.propTypes= {
    setCategories: PropTypes.func.isRequired
}

