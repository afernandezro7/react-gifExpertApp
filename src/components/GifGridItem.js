import React from 'react'

export const GifGridItem = ({id,title,url}) => {
    return (    
        <div className="card animate__animated animate__rollIn animate__delay-2s">
            <img src={ url } alt={ title } />
            <p>{ title }</p>
           
        </div>  
    )
}
