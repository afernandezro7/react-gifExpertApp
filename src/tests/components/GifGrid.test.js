import '@testing-library/jest-dom';
import React from 'react';
import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import {useFetchGifs} from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Pruebas sobre GifGrid', () => {
    
    const category = 'Hola Mundo'
    
    test('Debe mostrarse el componente ', () => {
        
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        })

        const wrapper = shallow(<GifGrid category = {category}/>);
        expect( wrapper ).toMatchSnapshot();
    });

    test('debe mostrar items cuando se cargan las img useFetch', () => {
        
        const gifs = [{
            id: 'abc',
            url: 'https://cualquiercosa.jpg',
            title: 'Goku'
        }]
        
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: true
        })
        
        const wrapper = shallow(<GifGrid category = {category}/>);
        
        expect( wrapper ).toMatchSnapshot();
    })
    
    
})



