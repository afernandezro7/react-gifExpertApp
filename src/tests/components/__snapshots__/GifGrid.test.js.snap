// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre GifGrid Debe mostrarse el componente  1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    Hola Mundo
     
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Cargando...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Pruebas sobre GifGrid debe mostrar items cuando se cargan las img useFetch 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    Hola Mundo
     
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Cargando...
  </p>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="abc"
      key="abc"
      title="Goku"
      url="https://cualquiercosa.jpg"
    />
  </div>
</Fragment>
`;
